General Fixes

shamoya opened the pull request.

shamoya committed:
mlx5_vdpa_pci_probe: Add return 0 at the end of the function

Currently, we enter the error label and release the priv.

Signed-off-by: Ido Shamay <idos@mellanox.com>

shamoya committed:
mlx5_vpda: Added some DEBUG logging in some error flows

Signed-off-by: Ido Shamay <idos@mellanox.com>

shamoya committed:
net/mlx5_vdpa: Implement get_queue_num callback

It returns max_num_virtqs from caps, which is currently
static from SW, until we have FW which support VDPA.

Also defined the general-purpose find_priv_resource_by_did, which
should be used from all vdpa_ops to get the priv for current
device from the global devices list.

Signed-off-by: Ido Shamay <idos@mellanox.com>

shamoya committed:
net/mlx5_vdpa: Implement get-features callback

Only the minimum for now - only VHOST protocol features support

Signed-off-by: Ido Shamay <idos@mellanox.com>

shahafsh commented on drivers/net/mlx5/mlx5_vdpa.c:
```diff
@@ -66,21 +66,33 @@ mlx5_vdpa_query_virtio_caps(struct vdpa_priv *priv)
 	MLX5_SET(query_hca_cap_in, in, op_mod,
 			(MLX5_HCA_CAP_GENERAL << 1) |
 			(MLX5_HCA_CAP_OPMOD_GET_CUR & 0x1));
+
```
in general the coding standard of Mellanox PMD is not to have blank lines apart from after the variable declaration.

if it helps for now we can have it, but will need to adjust when we go upstream. 
let me know.

shamoya replied:
Sure I'll fix.
Is this the convention throughout the DPDK project?
I'm regular from Linux kernel where need to add CR before/after if/while/for statements

shahafsh commented on drivers/net/mlx5/mlx5_vdpa.c:
```diff
@@ -66,21 +66,33 @@ mlx5_vdpa_query_virtio_caps(struct vdpa_priv *priv)
-                                       out, sizeof(out)))
-		return -1;
+                                       out, sizeof(out))) {
+	    DRV_LOG(ERR, "Failed to Query Current HCA CAP section\n");
```
same here. 

the error is something specific to mellanox usage, and not self explanatory for the general user. 
This is why it is better to keep those visible only under debug. 

if it helps for now, we can have it.

shamoya replied:
Sure, I'll change to debug

shahafsh commented on drivers/net/mlx5/mlx5_vdpa.c:
```diff
@@ -39,8 +43,48 @@ static struct vdpa_priv_list_head priv_list =
 static pthread_mutex_t priv_list_lock = PTHREAD_MUTEX_INITIALIZER;
 
+static struct vdpa_priv_list *
+find_priv_resource_by_did(int did)
```
maybe better to have helper to return the actual priv instead of the priv list?

shamoya replied:
I knida copied that from ifcvf, and I think it's more natural to return a linked-list node in a search function, rather then the inside, since the caller can traverse it from there (it it wants too).

shahafsh replied:
OK then.

shamoya commented:
Hi @shahafsh @bekets 

Did you have a chance to review? can this be merged?

shahafsh reviewed (commented):
Hi Ido,

find some small nits below.

shamoya commented:
@shahafsh @bekets 
Fixed the notes.
Also removed the get_protcol_features for now, saw with Liel yesterday it's not needed for now.
(until adding the doorbell feature)
